library/std/src/os/unix/net/ancillary.rs

```diff
@@ -642,3 +635,40 @@ impl<'a> SocketAncillary<'a> {
         self.truncated = false;
     }
 }
+
+/// An error returned when trying to add anciallary data that exceeds the buffer capacity.
+#[cfg(any(doc, target_os = "android", target_os = "linux",))]
+#[unstable(feature = "unix_socket_ancillary_data", issue = "76915")]
+pub struct AddAncillaryError {
```

Comment:
Should we really create an Error for on special failure? Maybe, create an enum `AncillaryError` for all errors, which can occur with Ancillary, or use `io:Error:other`.

Reply:
We could add an inner enum to change the error message, but I would be hestitent to expose the enum directly to the user. That would limit what we can do with it in the future, because we would have to stay backwards compatible.

I wouldn't directly return an `io::Error`, since it's a bit overkill to have a `Box<dyn Error>`. I kinda like an opaque cheap error type that can be converted into `io::Error`.

Reply:
Yes, `io::Error` would be okay, or we can use `#[non_exhaustive]` for the enum.